_studio/mfx_lib/scheduler/src/mfx_scheduler_core_ischeduler.cpp

```diff
@@ -683,5 +683,50 @@ mfxStatus mfxSchedulerCore::DoWork()
 } // mfxStatus mfxSchedulerCore::DoWork()
 
 
+mfxStatus mfxSchedulerCore::FindActiveSyncPoint(const void *pOwner, mfxSyncPoint *pSyncPoint)
+{
+    if (nullptr == pOwner || nullptr == pSyncPoint)
+    {
+        return MFX_ERR_NULL_PTR;
+    }
 
+    mfxTaskHandle handle;
+    handle.handle = 0;
+
+    mfxU32 priority;
+    // enter protected section
+    {
+        std::lock_guard<std::mutex> guard(m_guard);
+
+        for (priority = 0; priority < MFX_PRIORITY_NUMBER; priority += 1)
+        {
+            int type;
+
+            for (type = MFX_TYPE_HARDWARE; type <= MFX_TYPE_SOFTWARE; type += 1)
+            {
+                MFX_SCHEDULER_TASK *pTask = m_pTasks[priority][type];
+
+                // run over the tasks with particular priority
+                while (pTask)
+                {
+                    // return the syncPoint associated with the oldest working task which belongs to the provided owner
+                    if ((pTask->param.task.pOwner == pOwner) &&
+                        (MFX_WRN_IN_EXECUTION == pTask->opRes))
```

Comment:
Are we sure that actual tasks have only `MFX_WRN_IN_EXECUTION` status?

Reply:
It looks like 3 types of statuses are possible:
MFX_ERR_NONE, MFX_WRN_IN_EXECUTION and the rest(means errors). Should we return failed task here ? I think the answer is yes. @dvrogozh , what do you think?